launch/shell_unix.go

```diff
@@ -0,0 +1,71 @@
+// +build linux darwin
+
+package launch
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/api"
+)
+
+var (
+	bashCommandWithScript = `exec bash -c "$@"` // for processes w/o argsument
+	profileGlob           = "*"
+	appProfile            = ".profile"
+)
+
+func (l *Launcher) execWithShell(self string, process Process, profiles []string) error {
+	launcher := ""
+	for _, profile := range profiles {
+		launcher += fmt.Sprintf("source \"%s\"\n", profile)
+	}
+	bashCommand, err := l.bashCommand(process)
+	if err != nil {
+		return errors.Wrap(err, "determine bash command")
+	}
+	launcher += bashCommand
+	if err := l.Exec("/bin/bash", append([]string{
+		"bash", "-c",
+		launcher, self, process.Command,
```

Comment:
I recall there is some complexity around the use of `self` here and why it's needed. Is it possible we could document that as well, maybe in a comment?